toolchain/check/testdata/named_constraint/convert.carbon

```diff
@@ -0,0 +1,187 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/convert.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/named_constraint/convert.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/named_constraint/convert.carbon
+
+// --- type_and_with_named_constraint_self.carbon
+library "[[@TEST_NAME]]";
+
+constraint E {}
+
+// No constraints, so it matches any type, but you can't do anything with it.
+// Like `type`.
```

Comment:
Optional (here and in `compatible_constraints.carbon` below): is this comment needed? It focuses my attention on this line, which is kind of the wrong place for my attention to be, because IIUC the focus of this test is really on the call to `F` in `G`.